pkg/operator/starter.go

```diff
@@ -82,7 +74,8 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		versionGetter,
 		status.VersionForOperandFromEnv(),
 		controllerConfig.EventRecorder,
-		csiDriverConfigs)
+		csiDriverConfigs,
+		clusterOperatorStatus)
```

Comment:
I think it might be cleaner to write this code like:

```
	csiDriverController := csidriveroperator.NewCSIDriverStarterController(
		clients,
		resync,
		versionGetter,
		status.VersionForOperandFromEnv(),
		controllerConfig.EventRecorder,
		csiDriverConfigs)
	clusterOperatorStatus.WithRelatedObjectsFunc(csiDriverController.RelatedObjectFunc)
```

and in CSI driver starter:

```
func (c *CSIDriverStarterController) RelatedObjectFunc() func() (isset bool, objs []configv1.ObjectReference) {
	return func() (isset bool, objs []configv1.ObjectReference) {
		if len(c.relatedObjects) == 0 {
			return false, c.relatedObjects
		}
		return true, c.relatedObjects
	}
}
```

Reply:
I've updated the PR to take a similar approach. From initial testing, this works on AWS. I want to confirm other platforms (and test longer to ensure we're not flapping the relatedObjects).

Reply:
Just FYI - I've tested thi son a GCP cluster, and it includes the following relatedObject entry:

```
    {
      "group": "operator.openshift.io",
      "name": "pd.csi.storage.gke.io",
      "resource": "clustercsidrivers"
    },
```